IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -47,71 +47,122 @@ - (instancetype)init
     return self;
 }
 
-- (void)storeCredential:(MSIDCredentialCacheItem *)credential forKey:(MSIDCacheKey *)key
+- (void)storeItem:(id<MSIDJsonSerializable>)item inBucket:(NSString *)bucket forKey:(MSIDCacheKey *)key
 {
     dispatch_barrier_async(self.queue, ^{
-        [self.cacheObjects setObject:credential forKey:key];
+        
+        NSMutableDictionary *items = nil;
+        if (bucket && ![self.cacheObjects objectForKey:bucket])
+        {
+            items = [NSMutableDictionary new];
+        }
+        else
+        {
+            items = [self.cacheObjects objectForKey:bucket];
+        }
+        
+        [items setObject:item forKey:key];
+        [self.cacheObjects setObject:items forKey:bucket];
     });
 }
 
-- (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem
+- (void)mergeStorageItem:(MSIDMacCredentialStorageItem *)storageItem inBucket:(NSString *)bucket
 {
     dispatch_barrier_async(self.queue, ^{
-        for (MSIDCacheKey *key in storageItem.cacheObjects)
+        
+        for (NSString *bucketKey in storageItem.cacheObjects)
         {
-            MSIDCredentialCacheItem *credential = [storageItem.cacheObjects objectForKey:key];
-            if (credential)
+            NSMutableDictionary *bucketDict = [storageItem.cacheObjects objectForKey:bucketKey];
+            NSMutableDictionary *subDict = [self.cacheObjects objectForKey:bucketKey];
+            
+            if (!subDict)
             {
-                [self.cacheObjects setObject:credential forKey:key];
+                [self.cacheObjects setObject:bucketDict forKey:bucketKey];
             }
             else
             {
-                MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Credential is nil for key %@ while merging storage credentials.", MSID_PII_LOG_MASKABLE(key));
+                for (MSIDCacheKey *key in bucketDict)
```

Comment:
nit: what's about addEntriesFromDictionary instead of looping through entries ourselves?

Reply:
Addressed. This api was crashing for some unknown reasons so I decided to add my own merging logic. I could not reproduce the crash so updated code to use addEntriesFromDictionary.